package.json

```diff
@@ -20,14 +20,19 @@
     "@nomiclabs/hardhat-etherscan": "^2.1.6",
     "@nomiclabs/hardhat-truffle5": "^2.0.0",
     "@nomiclabs/hardhat-web3": "^2.0.0",
+    "@typechain/ethers-v5": "^8.0.5",
+    "@typechain/hardhat": "^3.1.0",
+    "@typechain/web3-v1": "^4.0.0",
     "@uma/common": "^2.14.0",
     "@uma/contracts-node": "^0.1.15",
     "@uniswap/v3-core": "^1.0.0-rc.2",
     "arb-ts": "^1.0.2",
     "decimal.js": "^10.3.1",
     "ethers": "^5.5.2",
     "glob": "^7.2.0",
-    "hardhat": "^2.6.8"
+    "hardhat": "^2.6.8",
+    "lodash.uniqby": "^4.7.0",
+    "typechain": "^6.0.0"
```

Comment:
we now have type chain in common, contracts-node, contracts-frontend and core. do all these locations need it given this change?

Reply:
It's a good point. Will double check if we can drop it everywhere outside of common. I think I've tried this before and had issues. I'll get to the bottom of it.

Reply:
So typechain isn't a dependency of common, and everywhere else it's a devDependency. This means that we don't "inherit" the installation from anywhere. I tried removing it and got an error message saying it was necessary.

Reply:
ye, that seems fine to leave it in that case. 

as a related point, do we know of any tools that can go through our lerna packages and look for ways to simplifying the dependency tree? i.e looking for packages we import and dont need and perhaps moving some stuff around to have fewer dependencies in each package?

Reply:
I don't! This would be useful, though! Would be interested if you have a suggestion.